Classes/GlobalStateExplorers/FLEXInstancesTableViewController.m

```diff
@@ -74,21 +80,93 @@ + (instancetype)instancesTableViewControllerForInstancesReferencingObject:(id)ob
     FLEXInstancesTableViewController *instancesViewController = [[self alloc] init];
     instancesViewController.instances = instances;
     instancesViewController.fieldNames = fieldNames;
+    [instancesViewController buildRowTitles];
     instancesViewController.title = [NSString stringWithFormat:@"Referencing %@ %p", NSStringFromClass(object_getClass(object)), object];
     return instancesViewController;
 }
 
++ (NSInteger)maxSections
+{
+    return 3;
+}
+
+- (NSPredicate *)predicateForSection:(NSInteger)section
+{
+    // These are the types of references that we typically don't care about.
+    // We want this list of "object-ivar pairs" split into two sections.
+    BOOL(^isObserver)(Pair, NSDictionary *) = ^BOOL(Pair pair, NSDictionary *bindings) {
+        NSString *row = pair[0];
+        return [row isEqualToString:@"__NSObserver object"] ||
+               [row isEqualToString:@"_CFXNotificationObjcObserverRegistration _object"];
+    };
+
+    /// These are common AutoLayout related references we also rarely care about.
+    BOOL(^isConstraintRelated)(Pair, NSDictionary *) = ^BOOL(Pair pair, NSDictionary *bindings) {
+        static NSSet *ignored = nil;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            ignored = [NSSet setWithArray:@[
+                @"NSLayoutConstraint _container",
+                @"NSContentSizeLayoutConstraint _container",
+                @"NSAutoresizingMaskLayoutConstraint _container",
+                @"MASViewConstraint _installedView",
+                @"MASLayoutConstraint _container",
+                @"MASViewAttribute _view"
+            ]];
+        });
+
+        NSString *row = pair[0];
+        return ([row hasPrefix:@"NSLayout"] && [row hasSuffix:@" _referenceItem"]) ||
+               ([row hasPrefix:@"NSIS"] && [row hasSuffix:@" _delegate"])  ||
+               ([row hasPrefix:@"_NSAutoresizingMask"] && [row hasSuffix:@" _referenceItem"]) ||
+               [ignored containsObject:row];
+    };
+
+    BOOL(^isEssential)(Pair, NSDictionary *) = ^BOOL(Pair pair, NSDictionary *bindings) {
+        return !(isObserver(pair, bindings) || isConstraintRelated(pair, bindings));
+    };
+
+    switch (section) {
+        case 0: return [NSPredicate predicateWithBlock:isEssential];
```

Comment:
Consider building a predicates array in `init` to remove the hardcoded `maxSections` count.